core-libs/setup/ssr/optimized-engine/optimized-ssr-engine.ts

```diff
@@ -285,7 +285,17 @@ export class OptimizedSsrEngine {
     }
 
     const renderingKey = this.getRenderingKey(request);
+
     const renderCallback: SsrCallbackFn = (err, html): void => {
+      /* Instead of providing another middleware for handling errors, we can handle it here.
+       * We still don't have access to the Config and OccEndpointsService so we still suffer from lack of information
+       * to check if we should map to CmsPageNotFoundServerErrorResponse
+       *
+       * IT IS RATHER BAD IDEA
+       *
+       * const mappedErr = mapErrorToServerErrorResponse(err);
+       */
+
```

Comment:
You said `IT IS RATHER BAD IDEA` but you mentioned it here.
So I'd like to understand what pros and cons do you see for this solution?